### ISSUE TITLE
Disable archive scanning from commandline

### ISSUE THREAD

**Issue by 0x736E** [trufflesecurity/trufflehog#2505]
## Description
Users should be able to enable or disable scanning file archives from the commandline.

There are situations where it is not desirable to scan file archives at all, and at present the only method to effectively disable this behaviour is by setting archive configurations which are impossible (e.g. --archive-max-size=1B).

### Preferred Solution
A new commandline flag such as --no-archive which disables the default behaviour of scanning file archives.

### Additional Context
N/A

#### References

* #884
* #1319
--->

**Comment by rgmz** [comment]
Related: #2257

**Comment by 0x736E** [comment]
Also related: #2506

**Comment by clonsdale-canva** [comment]
Yes please, somewhere a few versions ago archive scanning changed and it breaks consistently all my scans. I try the workaround to only scans archive with max size 1B but still doesn't succeed. I can't use the latest version of trufflehog until I can fully skip scanning all archives.

**Comment by 0x736E** [comment]
Can you please elaborate how your scans are still failing?

Perhaps also try setting the max timeout to 1ms:

--archive-max-timeout=1ms

**Comment by clonsdale-canva** [comment]
I haven't fully debugged the issue, since I'm running on GitHub Actions and it simply says it disconnects. Either the timeout or max size is ineffective. It could be chewing up too many resources to process the archives

**Comment by 0x736E** [comment]
@clonsdale-canva It sounds like you are enountering another issue in addition to the archive scanning issues. 

From my testing, archive scanning does not affect whether the scan succeeds or not, only the scope of what is scanned and the subsequent output as a result.

**Comment by dustin-decker** [comment]
We are working on plans to centralize archive handling which will make it easy to toggle on/off for all sources.

**Comment by clonsdale-canva** [comment]
@0x736E Spent some time debugging and you're correct. I am running trufflehog on many repositories at once in a highly parallel system. Some change in v3.64.0 caused it to tip over the edge with resource exhaustion which I mistakingly attributed to archive processing.

**Comment by brendan-wiz** [comment]
I would love a flag to just skip archives altogether.

**Comment by ahrav** [comment]
@dustin-decker, with all the recent changes to archive handling, I think we might be in a good position to support toggling archive scanning on/off holistically across all sources.

**Comment by clonsdale-canva** [comment]
Can this be closed now with the introduction of `--force-skip-archives`
